### ISSUE TITLE
Production build error with publishable angular libs and lazy loaded routes

### ISSUE THREAD

**Issue by aalbericio** [nrwl/nx#13532]
### Current Behavior

Using Nx latest version: 15.2.4

I have several lazy routes that are initialized inside a library (Library1 and Library2).

**Library 1** is a buildable (created from Nx tools) library

**Library 2** is a buildable and publishable (created from Nx tools) library 

Both libraries expose the same lazy loaded routes (A and B).

**Library 1** has been created with a project.json file that shows an executor: 

`"executor": "@nrwl/angular:ng-packagr-lite"`

**Library 2** has been created with a project.json file that shows an executor: 

`"executor": "@nrwl/angular:package"`

Now, let's use these modules:

In AppModule, we import our lazy routes from Library 1 and both serve and build work as they should: 

`imports: [
      BrowserModule,
      Lib1Module
  ],`

1. `serve `shows and serves our lazy loaded routes/modles
2. `build` produces all lazy loaded chunks as expected

Now, let's change our AppModule imports to import routes from Library 2:

`imports: [
      BrowserModule,
      Lib2Module
  ],`

The result is weird:

1. `serve `shows and serves our lazy loaded routes/modles
2. `build` throws an error:

`>  NX   Invalid value for option "output.file" - when building multiple chunks, the "output.dir" option must be used, not "output.file". To inline dynamic imports, set the "inlineDynamicImports" option.`

More on this:

If I then export those lazy loaded modules on Library2 index.ts like this:

export * from './lib/routes/a/route-a.module';
export * from './lib/routes/b/route-b.module';

Then our serve works as expected (showing and loading those lazy loaded modules fine) and **our build also works but it only produces a single chunk containing all modules! Effectively losing our lazy loaded features**.

### Expected Behavior

Both types of libraries should work (serve and build) the same way.

At this moment, I'm sticking to only-buildable Angular libs since they are working as I expect them to work. Moreover, I can publish them using npm publish.

You can play with the provided example/repository to reproduce this "bug?".

### Github Repo

https://github.com/aalbericio/nx-lazy

### Steps to Reproduce

Explained above.

### Nx Report

```shell
Not needed
```

**Comment by leosvelperez** [comment]
Thanks for reporting this!

## tldr;

This is not an issue caused by Nx. It's a limitation of `ng-packagr` that doesn't support lazy loaded routing in libraries.

## Full explanation

Lazy-loaded routing is not supported by `ng-packagr` as can be seen in the following issue https://github.com/ng-packagr/ng-packagr/issues/1960#issuecomment-827543254. The error that is thrown (as mentioned in this issue description) comes from Rollup. It happens when `ng-packagr` is creating some of the bundles specified in the [Angular Package Format (APF)](https://angular.io/guide/angular-package-format) and a dynamic import is found.

Both `@nrwl/angular:ng-packagr-lite` (buildable libs) and `@nrwl/angular:package` (publishable libs) executors invoke `ng-packagr` under the hood to build libraries and therefore, this limitation affects them.

However, as mentioned in this issue's description, lazy-loaded routes currently do work with buildable libraries. Why is that?

The `@nrwl/angular:ng-packagr-lite` executor, even though it invokes `ng-packagr`, strips out some functionalities that are not needed for libraries that are not going to be published (hence the name "buildable libraries") and therefore, don't need to produce an output that follows the APF. One of those stripped-out functionalities is the creation of extra bundles using Rollup. That's why the issue doesn't happen with buildable libraries.

The `@nrwl/angular:package` on the other hand, is meant to produce output that follows the APF and as such, it mirrors the functionality of the `@angular-devkit/build-angular:ng-packagr` (the Angular CLI builder powered by `ng-packagr`) with extra support on top for incremental builds. Because of that, the mentioned limitation applies to this one. You can try swapping `@nrwl/angular:package` with `@angular-devkit/build-angular:ng-packagr` and you'll get the same error. You'll also get the same error if you create an Angular CLI workspace with a similar setup.

## Some notes about other remarks in this issue description

> If I then export those lazy loaded modules on Library2 index.ts like this:
> 
> export * from './lib/routes/a/route-a.module'; export * from './lib/routes/b/route-b.module';
> 
> Then our serve works as expected (showing and loading those lazy loaded modules fine) and **our build also works but it only produces a single chunk containing all modules! Effectively losing our lazy loaded features**.

That's expected. By exporting the lazy-loaded module in the libraries, there's now a static import to it and therefore, it will be bundled together. So, as you mentioned, that's effectively losing the lazy loading feature.

> At this moment, I'm sticking to only-buildable Angular libs since they are working as I expect them to work. Moreover, I can publish them using npm publish.

Be mindful that the buildable library output is not meant to be used for publishing. It doesn't follow the APF specification, which is meant to be compatible with the tooling offered by the Angular team as well as the wider JavaScript ecosystem.

**Comment by aalbericio** [comment]
Oh! Thank you very much for  this elaborated answer. I really appreciate it.

Knowing this is an issue coming from the Angular framework itself, I'm gonna have to rethink shipping my lazy routes in a non 100% APF format or switching to a not-really-desired-in-my-scenario routing inside the main APP.

Thank you again for looking into this with so much detail.
